yucca/network_architectures/networks/UNet.py

```diff
@@ -231,8 +231,7 @@ def forward(self, x):
 
         x8 = torch.cat([self.upsample4(x7), x0], dim=1)
         x8 = self.decoder_conv4(x8)
-
-        if self.deep_supervision:
+        if self.deep_supervision and torch.is_grad_enabled():
```

Comment:
I would probably add a comment describing why we need the `is_grad_enabled` since it is not immediately obvious for everyone.